ivy/stateful/module.py

```diff
@@ -832,6 +830,27 @@ def train(self, mode: bool = True):
             if isinstance(module, ivy.Module):
                 module.train(mode=mode)
 
+    def to_device(self, device):
+        # moves the weights and buffers
+        # to the specified device
+
+        # moving weights and buffers to new device
+        for key, obj in self.v.items():
+            if (
+                isinstance(obj, ivy.Module)
+                or ivy.is_ivy_array(obj)
+                or isinstance(obj, ivy.Container)
+            ):
+                obj.to_device(device)
+            else:
+                ivy.to_device(obj, device=device)
+        if getattr(self, "buffers", None):
```

Comment:
I think we could refactor this further if `self.buffers` was a `{}` by default as we wouldn't need this check and we could also just instead add a `state_dict` and iterate over it instead of having 2 different loops for variables and buffers, what do you think?